### DISCUSSION TITLE
有没有可能发起一个项目基于OpenApi和Flurl.Http实现自动生成SDK?

### DISCUSSION THREAD

**Discussion by ZUOXIANGE** [fudiwei/DotNetCore.SKIT.FlurlHttpClient.Wechat#157] [Ideas]
这样像支付宝,openai等提供了openapi.yaml的服务都能获得像当前项目这样的使用体验了.

目前支付宝使用[openapi-generator](https://github.com/OpenAPITools/openapi-generator)生成的SDK并不好用.

可以使用[OpenAPI.NET](https://github.com/microsoft/OpenAPI.NET)解析openapi.yaml.

**Comment by fudiwei** [comment]
好想法，可以试试。

一开始我们也想过类似的方案，最后没能实施的主要原因是把官方文档里的内容转换成 OAS 或其他规范，这个工作量本身就已经比直接在 C# 里定义接口的工作量还要大了，而且很多接口里非标的东西太多了。

**Comment by ZUOXIANGE** [comment]
[alipay v3 openapi.yaml](https://github.com/alipay/alipay-sdk-java-all/blob/master/v3/api/openapi.yaml) 

[openai openapi.yaml](https://raw.githubusercontent.com/openai/openai-openapi/master/openapi.yaml)

这样的标准文档解析容易,主要是要有签名认证的之类的需要自定义.

发现一个这样的项目https://github.com/tryAGI/AutoSDK
